processor/memorylimiterprocessor/memorylimiter.go

```diff
@@ -81,6 +82,9 @@ type memoryLimiter struct {
 	configMismatchedLogged bool
 
 	obsrep *obsreport.Processor
+
+	lock       sync.Mutex
```

Comment:
To help clarify what the lock is for it can be useful to do this:
```suggestion
	refCounterLock       sync.Mutex
```

Reply:
👍